c/CMakeLists.txt

```diff
@@ -33,5 +33,50 @@ cmake_minimum_required(VERSION 3.10)
 # IMPORTED option, which only defines the targets within the scope of the library.
 # This allows the same target name to be used for different NVTX versions.
 
+# Project declaration needs to be before the configuration
+project(nvtx3 VERSION 3.1.1)
+
+# Add installation rules
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
 set(NVTX3_TARGETS_NOT_USING_IMPORTED ON)
 include(nvtxImportedTargets.cmake)
+
+
+set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_INCLUDEDIR}/nvtx3")
+
+# Configure the config file
+configure_package_config_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/nvtx3-config.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/nvtx3-config.cmake"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/nvtx3"
+    PATH_VARS INCLUDE_INSTALL_DIR
+)
+
+# Generate version file
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/nvtx3-config-version.cmake"
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
+)
+
+# Install headers
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/"
+        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
```

Comment:
We should comment on why we aren't using `INCLUDE_INSTALL_DIR` here

Reply:
I don't think I need `INCLUDE_INSTALL_DIR` anymore, it wasn't being used in `nvtx3-config.cmake.in` so I got rid of it. Is that an equally good solution?

Reply:
Looks great